pop-api/src/v0/fungibles/traits.rs

```diff
@@ -0,0 +1,116 @@
+use super::*;
+
+#[ink::trait_definition]
+pub trait Psp22 {
+	/// Returns the total token supply.
+	#[ink(message)]
+	fn total_supply(&self) -> Balance;
+
+	/// Returns the account balance for the specified `owner`.
+	///
+	/// # Parameters
+	/// - `owner` - The account whose balance is being queried.
+	#[ink(message)]
+	fn balance_of(&self, owner: AccountId) -> Balance;
+
+	/// Returns the allowance for a `spender` approved by an `owner`.
+	///
+	/// # Parameters
+	/// - `owner` - The account that owns the tokens.
+	/// - `spender` - The account that is allowed to spend the tokens.
+	#[ink(message)]
+	fn allowance(&self, owner: AccountId, spender: AccountId) -> Balance;
+
+	/// Transfers `value` amount of tokens from the caller's account to account `to`
+	/// with additional `data` in unspecified format.
+	///
+	/// # Parameters
+	/// - `to` - The recipient account.
+	/// - `value` - The number of tokens to transfer.
+	/// - `data` - Additional data in unspecified format.
+	#[ink(message)]
+	fn transfer(&mut self, to: AccountId, value: Balance, data: Vec<u8>) -> PSP22Result<()>;
```

Comment:
```suggestion
	fn transfer(&mut self, to: AccountId, value: Balance, data: Vec<u8>) -> Result<(), PSP22Error>;
```

Not sure about this one, would like to hear a second opinion here.

Reply:
I prefer `Result<(), PSP22Error>` to keep it aligned with the compliant: https://github.com/w3f/PSPs/blob/master/PSPs/psp-22.md#transfer_fromfrom-accountid-to-accountid-value-balance-data-u8--result-psp22error